lib/std/compress/zlib.zig

```diff
@@ -170,3 +257,19 @@ test "sanity checks" {
         testReader(&[_]u8{ 0x78, 0xda, 0x03, 0x00, 0x00 }, ""),
     );
 }
+
+test "compress data" {
+    const allocator = testing.allocator;
+    const rfc1951_txt = @embedFile("testdata/rfc1951.txt");
+
+    var compressed_data = std.ArrayList(u8).init(allocator);
+    defer compressed_data.deinit();
+
+    var compressor = try zlibStreamWriter(allocator, compressed_data.writer(), .{});
+    defer compressor.deinit();
+
+    try compressor.writer().writeAll(rfc1951_txt);
```

Comment:
1. This looks abit odd. Why is it called `zlibStreamWriter` if I need to use another `writer()`?
2. I would have expected, that `zlibStreamWriter` contains a `writeAll` method.

Reply:
Do you think `zlibStreamCompressor` would be a better name? I was just copying the already existing `zlibStreamReader`, which has a `reader()` method. If I changed it to `zlibStreamCompressor`, I should also change the existing function to `zlibStreamDecompressor` to match

Reply:
It sounds to me at least more consistent, since zlib streams can have [several modes like compress, decompress, deflate, inflate, gzip, gunzip](https://wiki.tcl-lang.org/page/zlib+stream).
See also https://github.com/madler/zlib/blob/master/deflate.c